Fixes labcoats having the wrong sprites when thrown

D4C-420 opened the pull request:
## About The Pull Request
Fixes #79450 


## Why It's Good For The Game
Direction sprites? For clothing on the floor? Crazy
## Changelog
:cl:
fix: lab coats no longer have directional sprites when thrown
/:cl:

D4C-420 committed:
fixes wrong pathing

changes pathing to match everything else

D4C-420 committed:
forgot the json

changed the json to point at the non worn file

D4C-420 committed:
Revert "fixes wrong pathing"

This reverts commit b3bb5f972a43cb2e155670aaef08c397ff2e9693.

Jolly-66 reviewed (commented):
I fixed this downstream, and again, is something that accidentally slipped by me when I was testing.

Make a clear `back` for obj sprites, it should resolve this runtime. Don't forget one for the adjusted version as well.

```
[19:16:16] Runtime in layer.dm,97: Configured icon state [back] was not found in icons/obj/clothing/suits/labcoat.dmi. Double check your json configuration.
  proc name: Initialize (/datum/greyscale_layer/icon_state/Initialize)
  src: /datum/greyscale_layer/icon_st... (/datum/greyscale_layer/icon_state)
  call stack:
  /datum/greyscale_layer/icon_st... (/datum/greyscale_layer/icon_state): Initialize('icons/obj/clothing/suits/labco...')
  /datum/greyscale_layer/icon_st... (/datum/greyscale_layer/icon_state): New('icons/obj/clothing/suits/labco...', /list (/list))
  Labcoat (/datum/greyscale_config/labcoat): ReadLayerGroup(/list (/list))
  Labcoat (/datum/greyscale_config/labcoat): ReadLayerGroup(/list (/list))
  Labcoat (/datum/greyscale_config/labcoat): ReadLayersFromJson(/list (/list))
  Labcoat (/datum/greyscale_config/labcoat): ReadIconStateConfiguration(/list (/list))
  Labcoat (/datum/greyscale_config/labcoat): Refresh(0)
  Greyscale (/datum/controller/subsystem/processing/greyscale): Initialize()
  Master (/datum/controller/master): init subsystem(Greyscale (/datum/controller/subsystem/processing/greyscale))
  Master (/datum/controller/master): Initialize(10, 0, 1)
[19:16:16] Runtime in layer.dm,97: Configured icon state [shoulder_t] was not found in icons/obj/clothing/suits/labcoat.dmi. Double check your json configuration.
```

D4C-420 committed:
Okay this should pass unit tests

Changes the paths for the json to the obj and adds in two empty sprites named back and back_t for the unit tests

D4C-420 commented:
> I fixed this downstream, and again, is something that accidentally slipped by me when I was testing.
Make a clear back for obj sprites, it should resolve this runtime. Don't forget one for the adjusted version as well.

thank you for the help did just that.

OrionTheFox commented on code/datums/greyscale/config_types/greyscale_configs/greyscale_clothes.dm:
```diff
@@ -360,8 +360,8 @@
-	icon_file = 'icons/mob/clothing/suits/labcoat.dmi'
-	json_config = 'code/datums/greyscale/json_configs/labcoat_worn.json'
+	icon_file = 'icons/obj/clothing/suits/labcoat.dmi'
+	json_config = 'code/datums/greyscale/json_configs/labcoat.json'
```
We should be able to cut down to only a single json_config (that was the hope with having different icon-files)

If you're willing to do that:
- Replace your current labcoat.json with this one - it has the missing Back component-states.
[labcoat.json](https://github.com/tgstation/tgstation/files/13236195/labcoat.json)

 - Then, remove the json_config define from greyscale_config/labcoat/worn (below this line) - it will inherit the json_config from here, and thus both the object and worn sprites can use the exact same set of code.

_Adding those blank 'back' states also fixed the only issue this would have caused too...
so really there were two solutions: add new icons and have them share a json, or set it to the other intended json. You basically did both fixes at once LMAO_

D4C-420 replied:
Did that and removed worn file

D4C-420 committed:
fixing unit tests

D4C-420 committed:
removed labcoat_worn.json

OrionTheFox commented on code/datums/greyscale/config_types/greyscale_configs/greyscale_clothes.dm:
```diff
@@ -360,14 +360,12 @@
 
 /datum/greyscale_config/labcoat/worn
 	name = "Labcoat (Worn)"
 	icon_file = 'icons/mob/clothing/suits/labcoat.dmi'
```
```suggestion
	icon_file = 'icons/mob/clothing/suits/labcoat.dmi'
	
```
everything checks out, just uh-
need to keep this whitespace between the config and the comment below (Can't post a Suggestion on deleted lines so I'm doing it here)

D4C-420 committed:
fixed line
